docs/features/string-literals-data-section.md

```diff
@@ -0,0 +1,18 @@
+# String literals in data section
+
+This opt-in Roslyn feature allows changing how string literals in C# programs are emitted into PE files (`.dll`/`.exe`).
+By default, string literals are emitted to the UserString heap which is limited to [2^24 bytes](https://github.com/dotnet/roslyn/issues/9852).
+When the limit is reached, the following compiler error is reported by Roslyn:
+
+```
+error CS8103: Combined length of user strings used by the program exceeds allowed limit. Try to decrease use of string literals.
+```
+
+By turning on the feature flag `utf8-string-encoding`, string literals (where possible) are instead emitted as UTF-8 data into a different section of the PE file
```

Comment:
>utf8-string-encoding [](http://example.com/codeflow?start=32&length=20)

"utf8-string-**literal**-encoding"?

Reply:
In the spec, I've renamed it to `data-section-string-literals` which IMO best reflects what the feature is about.